sql/src/main/java/com/vmware/ddlog/DDlogJooqProvider.java

```diff
@@ -558,26 +558,51 @@ private static DDlogRecord toValue(final Field<?> field, final Object in) {
         }
     }
 
-    private static void setValue(final Field<?> field, final DDlogRecord in, final Record out) {
-        final DataType<?> dataType = field.getDataType();
-        switch (dataType.getSQLType()) {
-            case Types.BOOLEAN:
-                out.setValue((Field<Boolean>) field, in.getBoolean());
-                return;
-            case Types.INTEGER:
-                out.setValue((Field<Integer>) field, in.getInt().intValue());
-                return;
-            case Types.BIGINT:
-                out.setValue((Field<Long>) field, in.getInt().longValue());
-                return;
-            case Types.VARCHAR:
-                out.setValue((Field<String>) field, in.getString());
-                return;
-            default:
-                throw new DDlogJooqProviderException("Unknown datatype " + field);
+    // Return a value based on the DDlog type of the record
+    private static Object ddlogRecordToObject(final Field<?> field, final DDlogRecord in) {
+        if (in.isBool()) {
+            return in.getBoolean();
+        } else if (in.isInt()) {
+            if (field.getDataType().getSQLType() == Types.BIGINT) {
+                return in.getInt().longValue();
+            } else {
+                return in.getInt().intValue();
+            }
+        } else if (in.isDouble()) {
+            return in.getDouble();
+        } else if (in.isFloat()) {
+            return in.getFloat();
+        } else if (in.isString()) {
+            return in.getString();
+        } else if (in.isStruct()) {
```

Comment:
how come that the original code didn't have this path?
were there no nulls so far in the tests?